daemons/fenced/fenced_commands.c

```diff
@@ -154,11 +154,17 @@ get_action_delay_base(stonith_device_t *device, const char *action, const char *
         char *value = strdup(hash_value);
         char *valptr = value;
 
+        CRM_ASSERT(value != NULL);
+
         if (victim) {
             for (char *val = strtok(value, "; \t"); val != NULL; val = strtok(NULL, "; \t")) {
                 char *mapval = strchr(val, ':');
 
-                if (strncmp(victim, val, (int)(mapval - val)) == 0) {
+                if (mapval == NULL || mapval[1] == 0) {
+                    crm_err("pcmk_delay_base: empty value in mapping", val);
+                }
+
+                if (strncasecmp(victim, val, (int)(mapval - val)) == 0) {
```

Comment:
if someone misconfigures a map (or map entry) starting with ':', then mapval==val, and strncasecmp(..., 0) will return 0 regardless of victim

also the third argument is size_t rather than int